src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -134,6 +134,21 @@ private void createTopic(final String topic, final int partitionsNum, final int
         } catch (final Exception e) {
             throw new TopicCreationException("Unable to create topic " + topic, e);
         }
+        // Next step is to wait for topic initialization. On can not skip this task, cause kafka instances may not
+        // receive information about topic creation, which in turn will block publishing.
+        // This kind of behavior was observed during tests, but may also present on highly loaded event types.
+        final long timeoutMillis = TimeUnit.SECONDS.toMillis(5);
+        final long borderTime = System.currentTimeMillis() + timeoutMillis;
+        boolean creationConfirmed = false;
+        while (!creationConfirmed && borderTime >= System.currentTimeMillis()) {
+            try (Consumer<byte[], byte[]> consumer = kafkaFactory.getConsumer()) {
+                final List<PartitionInfo> partitions = consumer.partitionsFor(topic);
+                creationConfirmed = null != partitions;
+            }
+        }
```

Comment:
1. Ok, I will do what you propose. 
2. Added interval of 100 ms
3. Because I am not sure, that behavior of producer and consumer is the same, so I decided to use exactly the same call that was failing.